synapse/python_dependencies.py

```diff
@@ -107,7 +107,7 @@
     "systemd": ["systemd-python>=231"],
     "url_preview": ["lxml>=3.5.0"],
     "sentry": ["sentry-sdk>=0.7.2"],
-    "opentracing": ["jaeger-client>=4.0.0", "opentracing>=2.2.0"],
+    "opentracing": ["jaeger-client>=4.0.0,<4.7.0", "opentracing>=2.2.0"],
```

Comment:
[The 4.7 release](https://github.com/jaegertracing/jaeger-client-python/releases/tag/4.7.0) [requires Python 3.7+](https://github.com/jaegertracing/jaeger-client-python/pull/325), and I think we're still supporting 3.6, so this is a no-go anyway. Might be worth leaving a comment to explain why we need that max version?

As for whypy mypy fails, I'm not sure. [They run mypy as part of `make lint`](https://github.com/jaegertracing/jaeger-client-python/blob/34922a8546620045151756b583b3c201e4929601/Makefile#L72), which in turn gets run by `make test_ci` and that is run by GHA.

Maybe the two projects have different mypy config?

Reply:
> [The 4.7 release](https://github.com/jaegertracing/jaeger-client-python/releases/tag/4.7.0) [requires Python 3.7+](https://github.com/jaegertracing/jaeger-client-python/pull/325)

If they have their package setup correctly Python 3.6 should install <4.7.0 anyway...

I'll add a comment though.

Reply:
> As for whypy mypy fails, I'm not sure. [They run mypy as part of `make lint`](https://github.com/jaegertracing/jaeger-client-python/blob/34922a8546620045151756b583b3c201e4929601/Makefile#L72), which in turn gets run by `make test_ci` and that is run by GHA.
> 
> Maybe the two projects have different mypy config?

I'm confused by this -- I suspect our mypy is failing since they added type hints which mypy is now finding? But I'm not 100% sure.

Reply:
Ahh, I didn't realise the rust reporter was our own thing!